sportsref/ncaaf/players.py

```diff
@@ -157,8 +171,66 @@ def defense(self):
         :returns: Pandas DataFrame with defensive stats.
         """
         doc = self.getDoc()
-        table = doc('#defense')
+        table = doc('#all_defense')
         df = sportsref.utils.parseTable(table)
+        if df.empty and table.length > 0:
+            tree = etree.fromstring(str(table))
+            comments = tree.xpath('//comment()')
+            comment = etree.tostring(comments[0])
+            contents = comment.replace("<!--", "").replace("-->", "")
+            table = pq(contents)
+            df = sportsref.utils.parseTable(table)
+        return df
+
+    @sportsref.decorators.memoized
+    def scoring(self):
+        """Gets yearly scoring stats for the player.
+        :returns: Pandas DataFrame with defensive stats.
+        """
+        doc = self.getDoc()
+        table = doc('#all_scoring')
+        df = sportsref.utils.parseTable(table)
+        if df.empty and table.length > 0:
+            tree = etree.fromstring(str(table))
+            comments = tree.xpath('//comment()')
+            comment = etree.tostring(comments[0])
+            contents = comment.replace("<!--", "").replace("-->", "")
+            table = pq(contents)
+            df = sportsref.utils.parseTable(table)
+        return df
+
+    @sportsref.decorators.memoized
+    def punt_kick_returns(self):
+        """Gets yearly scoring stats for the player.
+        :returns: Pandas DataFrame with defensive stats.
+        """
+        doc = self.getDoc()
+        table = doc('#all_punt_ret')
```

Comment:
Looks to me like the right ID here is '#kick_ret' - based on [Artavis Scott's page](http://www.sports-reference.com/cfb/players/artavis-scott-1.html)